dig_test.go

```diff
@@ -700,6 +700,49 @@ func TestEndToEndSuccess(t *testing.T) {
 			assert.Nil(t, p.Baz, "Baz must be unset")
 		}), "invoke failed")
 	})
+
+	t.Run("variadic arguments invoke", func(t *testing.T) {
+		c := New()
+
+		type A struct{ i int }
+
+		require.NoError(t, c.Provide(func() []*A {
+			return []*A{{1}, {2}, {3}}
+		}), "failed to provide A slice")
+
+		require.NoError(t, c.Invoke(func(as ...*A) {
```

Comment:
variadic types are different from slices and I don't think that we should mix them.
Consider this case: type is missing from the graph and we invoke a variadic function that expands this type. There should be no error, right? I expect this test to pass:
```golang
t.Run("alsam", func(t *testing.T) {
	c := New()
	require.NoError(t, c.Invoke(func(n ...int) {
	assert.Empty(t, n)
}), "failed to invoke")
```